Agoda.Frameworks.DB/DbRepository.cs

```diff
@@ -60,11 +60,12 @@ protected virtual ShouldRetryPredicate ShouldRetry(int maxAttemptCount)
         private TFuncResult ExecuteCacheOrGet<TRequest, TResult, TFuncResult>(
             IStoredProc<TRequest, TResult> sp,
             TRequest parameters,
-            Func<TFuncResult> getResultFunc)
+            Func<TFuncResult> getResultFunc,
+            string cacheKey = "")
         {
             return EnableCache(sp)
-                ? _cache.GetOrCreate(
-                    sp.GetParameters(parameters).CreateCacheKey(sp.StoredProcedureName),
+                ? _cache.GetOrCreate(string.IsNullOrEmpty(cacheKey) ?
+                    sp.GetParameters(parameters).CreateCacheKey(sp.StoredProcedureName) : cacheKey,
```

Comment:
What happens when the store proc parameter is a session id? IP address? etc. are we concerned about this?

Reply:
No, I am not concern about that because It is the responsibility of the client code